projects/rear_power_distribution/src/rear_power_distribution_current_measurement.c

```diff
@@ -0,0 +1,122 @@
+#include "rear_power_distribution_current_measurement.h"
+
+#include <stddef.h>
+#include "bts_7200_current_sense.h"
+#include "mcp23008_gpio_expander.h"
+#include "sn74_mux.h"
+
+#define DSEL_MCP23008_I2C 0x20
+
+// These are global constants declared in the header file
+const GpioAddress REAR_POWER_DISTRIBUTION_CURRENT_I2C_SDA_ADDRESS = { .port = GPIO_PORT_B,
+                                                                      .pin = 11 };
+const GpioAddress REAR_POWER_DISTRIBUTION_CURRENT_I2C_SCL_ADDRESS = { .port = GPIO_PORT_B,
+                                                                      .pin = 10 };
+
+typedef enum {
+  // first thing named is when DSEL is 0, then when DSEL is 1
+  REAR_POWER_DISTRIBUTION_BTS7200_CTR_BRK_STROBE = 0,
+  REAR_POWER_DISTRIBUTION_BTS7200_LEFT_RIGHT_BRK_LIGHT,
+  REAR_POWER_DISTRIBUTION_BTS7200_LEFT_RIGHT_TURN,
+  REAR_POWER_DISTRIBUTION_BTS7200_SOLAR_TELEMETRY,
+  // there's also a CAM_CHARGER bts7200 on the rear board (?)
+  NUM_REAR_POWER_DISTRIBUTION_BTS7200,
+} RearPowerDistributionBts7200;
+
+static const Mcp23008GpioAddress s_bts7200_addresses[] = {
+  [REAR_POWER_DISTRIBUTION_BTS7200_CTR_BRK_STROBE] = { .i2c_address = DSEL_MCP23008_I2C, .pin = 1 },
+  [REAR_POWER_DISTRIBUTION_BTS7200_LEFT_RIGHT_BRK_LIGHT] = { .i2c_address = DSEL_MCP23008_I2C,
+                                                             .pin = 2 },
+  [REAR_POWER_DISTRIBUTION_BTS7200_LEFT_RIGHT_TURN] = { .i2c_address = DSEL_MCP23008_I2C,
+                                                        .pin = 3 },
+  [REAR_POWER_DISTRIBUTION_BTS7200_SOLAR_TELEMETRY] = { .i2c_address = DSEL_MCP23008_I2C,
+                                                        .pin = 7 },
+};
+
+static const uint8_t s_bts7200_to_mux_sel[] = {
+  [REAR_POWER_DISTRIBUTION_BTS7200_CTR_BRK_STROBE] = 2,
+  [REAR_POWER_DISTRIBUTION_BTS7200_LEFT_RIGHT_BRK_LIGHT] = 3,
+  [REAR_POWER_DISTRIBUTION_BTS7200_LEFT_RIGHT_TURN] = 0,
+  [REAR_POWER_DISTRIBUTION_BTS7200_SOLAR_TELEMETRY] = 7,
+};
+
+// again, rear power distribution mux address
+static const Sn74MuxAddress s_mux_address = {
+  .sel_pins =
+      {
+          { .port = GPIO_PORT_A, .pin = 6 },  //
+          { .port = GPIO_PORT_A, .pin = 5 },  //
+          { .port = GPIO_PORT_A, .pin = 4 },  //
+      },
+  .mux_output_pin = { .port = GPIO_PORT_B, .pin = 0 },  //
+};
+
+static RearPowerDistributionCurrentStorage s_storage = { 0 };
```

Comment:
too many static variables, would you be able to typedef a struct:`RearPowerDistributionCurrentMeasurement` and have all of these be the members of that struct? Then have the user pass that in as storage... chances are we'll reuse this module for both front and rear power dist.